src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2742,4 +2743,163 @@ namespace winrt::TerminalApp::implementation
             WindowRenamer().IsOpen(false);
         }
     }
+
+    // - Stop previewing the currently previewed action. We can use this to
+    //   clean up any state from that action's preview.
+    // - We use _lastPreviewedCommand to determine what type of action to clean up.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TerminalPage::_EndPreview()
+    {
+        if (_lastPreviewedCommand == nullptr || _lastPreviewedCommand.Action() == nullptr)
+        {
+            return;
+        }
+        switch (_lastPreviewedCommand.Action().Action())
+        {
+        case ShortcutAction::SetColorScheme:
```

Comment:
Hmmmm I sort of wanted a throw/failfast on the default case here so we could catch when someone adds another preview action but forgets to clean it up since that's no good.

But the more I read... it looks like preview and end preview are always being called for actions and the ones that don't support it just fall through, yeah?

Reply:
Yeppers.